packages/odyssey-react-mui/src/labs/DataComponents/TableContent.tsx

```diff
@@ -0,0 +1,384 @@
+    }) => void;
+    handleDragHandleOnDragCapture: (
+      table: MRT_TableInstance<MRT_RowData>,
+    ) => void;
+    handleDragHandleOnDragEnd: (table: MRT_TableInstance<MRT_RowData>) => void;
+    resetDraggingAndHoveredRow: (table: MRT_TableInstance<MRT_RowData>) => void;
+    updateRowOrder: ({
+      rowId,
+      newRowIndex,
+    }: {
+      rowId: string;
+      newRowIndex: number;
+    }) => void;
+  };
+  rowSelection: MRT_RowSelectionState;
+  setRowSelection: Dispatch<SetStateAction<MRT_RowSelectionState>>;
+  setTableState: Dispatch<SetStateAction<TableState>>;
+  tableOptions: TableProps;
+  tableState: TableState;
+  totalRows: UniversalProps["totalRows"];
+};
+
+const TableContent = ({
+  columns,
+  data,
+  draggingRow,
+  emptyState,
+  getRowId,
+  hasRowReordering,
+  hasRowSelection,
+  isEmpty,
+  isLoading,
+  isNoResults,
+  isRowReorderingDisabled,
+  onReorderRows,
+  pagination,
+  rowReorderingUtilities,
+  rowSelection,
+  setRowSelection,
+  setTableState,
+  tableOptions,
+  tableState,
+  totalRows,
+}: TableContentProps) => {
+  const [isTableContainerScrolledToStart, setIsTableContainerScrolledToStart] =
+    useState(true);
+  const [isTableContainerScrolledToEnd, setIsTableContainerScrolledToEnd] =
+    useState(true);
+  const [tableInnerContainerWidth, setTableInnerContainerWidth] =
+    useState<string>("100%");
+  const tableOuterContainerRef = useRef<HTMLDivElement>(null);
+  const tableInnerContainerRef = useRef<HTMLDivElement>(null);
+  const tableContentRef = useRef<HTMLTableElement>(null);
+
+  useScrollIndication({
+    setIsTableContainerScrolledToEnd: setIsTableContainerScrolledToEnd,
+    setIsTableContainerScrolledToStart: setIsTableContainerScrolledToStart,
+    setTableInnerContainerWidth: setTableInnerContainerWidth,
+    tableInnerContainer: tableInnerContainerRef.current,
+    tableOuterContainer: tableOuterContainerRef.current,
+  });
+
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const { t } = useTranslation();
+
+  const columnIds = useMemo(() => {
+    return columns.map((column) => column.accessorKey) ?? [];
+  }, [columns]);
+
+  const columnOrder = useMemo(
+    () => [
+      "mrt-row-drag",
+      "mrt-row-select",
+      "mrt-row-expand",
+      ...columnIds,
+      "mrt-row-actions",
+    ],
+    [columnIds],
+  ) as string[];
+
+  const rowDensityClassName = useMemo(() => {
+    return tableState.rowDensity === "spacious"
+      ? "MuiTableBody-spacious"
+      : tableState.rowDensity === "compact"
+        ? "MuiTableBody-compact"
+        : "MuiTableBody-default";
+  }, [tableState]);
+
+  const defaultCell = useCallback(
+    ({ cell }: { cell: DataTableCell<MRT_RowData> }) => {
+      const value = cell.getValue<string>();
+      const hasTextWrapping =
+        cell.column.columnDef.hasTextWrapping ||
+        cell.column.columnDef.enableWrapping;
+
+      return hasTextWrapping ? value : <TextWrapper>{value}</TextWrapper>;
+    },
+    [],
+  );
```

Comment:
I think we want this, but I'm not sure. It's been a while since I've done any React, so I'd look up if it's an issue to use `useCallback` without passing anything as generic:
```suggestion
  const defaultCell = useCallback<({ cell }: { cell: DataTableCell<MRT_RowData> }) => ReactElement | typeof value>(
    ({ cell }) => {
      const value = cell.getValue<string>();
      const hasTextWrapping =
        cell.column.columnDef.hasTextWrapping ||
        cell.column.columnDef.enableWrapping;

      return hasTextWrapping ? value : <TextWrapper>{value}</TextWrapper>;
    },
    [],
  );
```